### PR TITLE
SMS compliance flow fix

### PR DESCRIPTION
**Pull request by lidio601**
Here are the fixes as suggested in https://trello.com/c/aNqpwqze/547-smsflowhandlereturnstate-logic-error

### CODE REVIEW

File: lib/brain.js [hunk 1]
```diff
@@ -1251,6 +1251,7 @@ Brain.prototype.isTierCompliant = function isTierCompliant (tier) {
     case 'sms':
       return !_.isNil(tx.phone)
     case 'hardLimit':
+      this.failedCompliance = 'hardLimit'
```
Comment (lidio601) [inline lib/brain.js]: In think this is not right because it may override other failed compliance

File: lib/brain.js [hunk 2]
```diff
@@ -1267,10 +1267,12 @@ Brain.prototype.smsFlowHandleReturnState = function smsFlowHandleReturnState ()
...
   }
 
-  const amount = this.getUsersDailyVolume().add(this.lastRejectedBillFiat)
+  const amount = this.complianceDailyVolumeAmount()
   const nonCompliant = this.nonCompliantTiers(amount)
 
-  const otherTiers = _.isNil(this.failedCompliance) && !_.isEmpty(nonCompliant)
+  const otherTiers = _.isNil(this.failedCompliance) &&
+    !_.isEmpty(nonCompliant) &&
+    _.head(nonCompliant) !== 'hardLimit'
```
Comment (lidio601) [inline lib/brain.js]: I moved to this check instead to skip the `return complianceTiers.run(_.head(nonCompliant))` with _hardLimit_ that would crash

**Comment by joshmh** [comment]
Thanks @lidio601, might take me a bit longer to review since I really need to remember how this code works.

File: lib/brain.js (hunk 2 above)
Comment (joshmh) [reply lib/brain.js]: But if there's a hardLimit, won't it just skip compliance? Where will it stop the tx if the user has exceeded hardLimit?

Comment (lidio601) [reply lib/brain.js]: This currently throw [here](https://github.com/lamassu/lamassu-machine/blob/v3/lib/compliance-tiers.js#L40)

File: lib/brain.js [hunk 3]
```diff
@@ -1283,17 +1285,21 @@ Brain.prototype.minimumFiat = function minimumFiat () {
...
+Brain.prototype.complianceDailyVolumeAmount = function complianceDailyVolumeAmount () {
   const tx = this.tx
   const minimumTx = tx.direction === 'cashIn'
     ? _.defaultTo(BN(0), tx.minimumTx)
     : this.minimumFiat()
 
-  const amount = this.getUsersDailyVolume()
+  return this.getUsersDailyVolume()
     .add(this.lastRejectedBillFiat)
     .add(minimumTx)
```
[the remark this answers is not in the document]
Comment (lidio601) [reply lib/brain.js]: sorry I can't see what do you mean. at 1295 we're adding minimumTx ok, but I can't see where we're considering the current tx balance.

File: lib/brain.js (hunk 2 above)
Comment (joshmh) [reply lib/brain.js]: Right but that's never supposed to happen. Hitting the hardLimit is supposed to happen and shouldn't throw an error, it should make the user complete the transaction and not allow any more bills.

**Comment by lidio601** [comment]
sorry it's not easy to see in "files changed" tab.

I moved complianceReturnState and smsFlowHandleReturnState since the first were only called by smsFlowHandleReturnState.
Now they use the same method to compute the _complianceAmount_.

I'm not sure whenever the complianceAmount is correct, btw now the consider the same limit.

**Comment by joshmh** [comment]
@lidio601 Looks like this has conflicting changes. Can you rebase on top of latest dev?

**Comment by lidio601** [comment]
done =)

**Comment by joshmh** [comment]
Thanks! Added a couple more comments.

**Pull request merged**